### ISSUE TITLE
Cannot read property 'curve' of undefined

### ISSUE THREAD

**Issue by VladislavIvanov** [aws-samples/non-profit-blockchain#31]
Hi, @MCLDG !

I have started the non-profit-blockchain example project for about a week. 

Today I checked the project and I found that something went wrong with the users certificates. Each request to the network returns:
```
##### getRegisteredUser - Got admin property [ { username: 'admin', secret: 'Adminpwd1!' } ]
[FileKeyValueStore.js]: getValue { key: 'admin' }
[crypto_ecdsa_aes]: importKey - start
[crypto_ecdsa_aes]: importKey - have the key [Circular]
[utils.CryptoKeyStore]: This class requires a CryptoKeyStore to save keys, using the store: {"opts":{"path":"/tmp/fabric-client-kv-org1"}}
[FileKeyValueStore.js]: constructor { options: { path: '/tmp/fabric-client-kv-org1' } }
[utils.CryptoKeyStore]: _getKeyStore returning ks
[FileKeyValueStore.js]: getValue { key: '8f59ca67beed7dbdfe431df07ce5b2bb900088d118d37a3425af91f8e5f84a61-priv' }
[FileKeyValueStore.js]: getValue { key: '8f59ca67beed7dbdfe431df07ce5b2bb900088d118d37a3425af91f8e5f84a61-pub' }
error: [Client.js]: Failed to load user "admin" from local key value store. Error: TypeError: Cannot read property 'curve' of undefined
##### getRegisteredUser - Failed to get registered user: user101 with error: TypeError: Cannot read property 'curve' of undefined

```

I thought that the certificates for admin and user1(for example) are expired, but after I checked the list of certificates I had marked that they should not have expired:
```
        Issuer: C=US,ST=Washington,UnknownOID=2.5.4.7,O=Amazon Web Services, Inc.,OU=Amazon Managed Blockchain,CN=member Amazon Managed Blockchain Root CA
        Validity
            Not Before: Jun 20 17:40:00 2019 UTC
            Not After : Jun 17 17:45:00 2029 UTC
        Subject: C=US,ST=North Carolina,O=Hyperledger,OU=client,OU=member,CN=admin
```

Can you give me an advice how to resolve this issue? 
Thanks!

**Comment by MCLDG** [comment]
You have an old copy of a cert somewhere. Possibly in your msp directory or
in the temp kv store if the error appears in the ui.

**Comment by VladislavIvanov** [comment]
Hi Michael,

I have two certificates in the ```fabric-client-kv-org1``` - ```admin``` and ```user```.  Their validity is till 2029.

- What do you mean with an old copy? 
- Do you think it could be because of some caching. 
The application works as expected when the one interacts/tests the network every day, but when no one is using the network for a few days, the next network usage is accompanied with the mistake of not discovering certificates in the crypto-store, which provokes the error - ```Cannot read property 'curve' of undefined```.

Thanks!

**Comment by MCLDG** [comment]
I see this error if I generate a new admin cert in the msp directory for
example.  In this case the msp dir contains both the old and new cert, and
the error appears if fabric picks the wrong cert.

For the client app, I'm not sure what is happening in your case. If you
rename the kv store directory does the problem go away? I assume you are
not registering a new user. I've only seen this very helpful error if the
certs are different to what is expected.

**Comment by VladislavIvanov** [comment]
The is no duplicating of the admin cert. The same certificates ```admin```, ```user``` are located in the crypto store, but the fabric can't pick the right admin cert.

The renaming/deleting of the kv works.
My goal is to figure out how to resolve the manual renaming of the crypto store.

The user is created with a call to the ```/users```.
I wasn't able to understand when an admin cert is being created?

**Comment by ccchhheeemmmaaa** [comment]
Same error... All structures re-uped and same error throwing...
Please Can you review the code for this issue created 7 days ago?
Thanks in advance...

**Comment by ccchhheeemmmaaa** [comment]
Hi, today is working!

**Comment by VladislavIvanov** [comment]
> Hi, today is working!

Hey @ccchhheeemmmaaa! 

Did you do any specific actions to resolve the error(e.g. rename the crypto store) or it worked on its own?

Thanks!

**Comment by ccchhheeemmmaaa** [comment]
Hi! terminate the vpc and reup... is not the better solution and it's bad
in production environment. We need to knows what is the real problem &
solution

**Comment by VladislavIvanov** [comment]
I totally agree. 

@MCLDG, please help with this issue. 
Renaming of the crypto store and rebuilding of the VPC are not sustainable long-term solutions. 

Thanks in advance! ✌️

**Comment by ebaizel** [comment]
This may be due to the `/tmp` folder being deleted and the keys not being found.  This could happen if the EC2 instance were to be restarted.  Can you see the keys for `user101` in `/tmp`?

When you register/enroll via the API, the crypto store is set as `/tmp/`, and the setting is configured in `ngo-rest-api/connection-profile/client-org1.yaml`.  You could try updating the path to something outside of `/tmp` so it persists across restarts.

**Comment by kutsuna-m** [comment]
@ebaizel 
Thank you for your information!
I met same issue. In order to your advice, I could solve this problem.

**Comment by ebaizel** [comment]
Thanks for the update @kutsuna-m.  Glad to hear it worked for you.

**Comment by JMelloni** [comment]
@kutsuna-m @ebaizel , Hi there, have been working on an Amazon Managed Blockchain project for a few weeks, but just now came across this issue. Once you change the storage path to a non /tmp path, what steps do you take? Thanks!

**Comment by ccchhheeemmmaaa** [comment]
@JMelloni after rename certificate storage allocation outside root/tmp only it's required change the path in Client-org1.yaml in connection profile of api (remember that file is in non-profit / connection-prof because is a copy of template) and works... Linux v14 to up remove root/tmp directory

**Comment by ebaizel** [comment]
@JMelloni if you've already registered and enrolled users, you'll need to copy over the certificates from /tmp into your new folder, and then update your config files to point to this new folder.  and then restart any fabric clients you may have running.

alternatively, if you have not enrolled and registered any users, you can simply update your config files to point to this new folder, and then restart any fabric clients.  hope that helps.

**Comment by JMelloni** [comment]
Thanks for the quick responses!

**Comment by MCLDG** [comment]
This issue should be resolved by #42 and #48

**Comment by SHIVAMGUPTA24** [comment]
shivam@shivam-HP-Notebook:~/project final year/pharmaceutical-supply-chain/pharma-chain$ node initTransaction.js 
{"selector":{}}
Error processing transaction. TypeError: Cannot read property 'curve' of undefined
TypeError: Cannot read property 'curve' of undefined
    at Object.KEYUTIL.getKey (/home/shivam/project final year/pharmaceutical-supply-chain/pharma-chain/node_modules/jsrsasign/lib/jsrsasign.js:247:10379)
    at CryptoSuite_ECDSA_AES.getKey (/home/shivam/project final year/pharmaceutical-supply-chain/pharma-chain/node_modules/fabric-client/lib/impl/CryptoSuite_ECDSA_AES.js:206:27)
    at <anonymous>
Disconnect from Fabric gateway.
done
